crates/chaincash_transactions/src/reserves.rs

```diff
@@ -0,0 +1,113 @@
+use crate::TransactionBuilderError;
+use ergo_lib::chain::ergo_box::box_builder::ErgoBoxCandidateBuilder;
+use ergo_lib::chain::transaction::unsigned::UnsignedTransaction;
+use ergo_lib::ergo_chain_types::EcPoint;
+use ergo_lib::ergotree_ir::chain::address::NetworkAddress;
+use ergo_lib::ergotree_ir::chain::ergo_box::ErgoBox;
+use ergo_lib::ergotree_ir::chain::{ergo_box::NonMandatoryRegisterId, token::Token};
+use ergo_lib::ergotree_ir::ergo_tree::ErgoTree;
+use ergo_lib::wallet::{box_selector::BoxSelection, tx_builder::TxBuilder};
+
+// #[derive(Serialize, Deserialize, Clone)]
+// pub struct MintReserveRequest {
+//     pub public_key_hex: String,
+//     pub amount: u64,
+// }
+//
+// pub fn mint_reserve_transaction(
+//     request: MintReserveRequest,
+//     reserve_tree: ErgoTree,
+//     inputs: BoxSelection<ErgoBox>,
+//     context: TxContext,
+// ) -> Result<UnsignedTransaction, TransactionError> {
+//     let pk = EcPoint::try_from(request.public_key_hex).map_err(TransactionError::Parsing)?;
+//     let mut reserve_box_builder = ErgoBoxCandidateBuilder::new(
+//         request.amount.try_into()?,
+//         reserve_tree,
+//         context.current_height,
+//     );
+//     let nft_id = inputs
+//         .boxes
+//         .get(0)
+//         .ok_or_else(|| {
+//             TransactionError::MissingBox(
+//                 "mint_reserve_transaction: failed to find input box required to mint nft"
+//                     .to_string(),
+//             )
+//         })?
+//         .box_id();
+//     let nft = Token {
+//         token_id: nft_id.into(),
+//         amount: 1.try_into()?,
+//     };
+//     reserve_box_builder.add_token(nft);
+//     reserve_box_builder.set_register_value(NonMandatoryRegisterId::R4, pk.into());
+//     Ok(TxBuilder::new(
+//         inputs,
+//         vec![reserve_box_builder.build()?],
+//         context.current_height,
+//         context.fee.try_into()?,
+//         NetworkAddress::try_from(context.change_address)?.address(),
+//     )
+//     .build()?)
+// }
+
+pub fn mint_nft_from_inputs(inputs: &[ErgoBox]) -> Token {
+    mint_token_from_inputs(inputs, 1)
+}
+
+pub fn mint_token_from_inputs(inputs: &[ErgoBox], amount: u64) -> Token {
+    let token_id = inputs.get(0).unwrap().box_id();
+    Token {
+        token_id: token_id.into(),
+        amount: amount.try_into().unwrap(),
+    }
+}
+
+#[derive(Default)]
+pub struct MintReserveTransactionBuilder {
+    owner: Option<String>,
+    amount: Option<u64>,
+    height: Option<u32>,
+    inputs: Option<Vec<ErgoBox>>,
+    change_address: Option<String>,
+    // TODO: use a constant, reserve contracts dont need to be compiled
+    ergo_tree: Option<ErgoTree>,
+}
+
+impl MintReserveTransactionBuilder {
```

Comment:
Builder pattern doesnt seem worth it, basically everything is the same anyway except more boiler plate to unwrap stuff. Probably just keep with original standalone functions but maybe rethink function args to make it more convinient